hoomd/mpcd/SphereGeometry.h

```diff
@@ -55,26 +56,47 @@ class __attribute__((visibility("default"))) SphereGeometry
             {
 
             /*
-             * If particle is still inside the sphere or has zero speed, no collision could have occurred and therefore
-             * exit immediately.
+             * If particle is still inside the sphere , no collision could have occurred and therefore
+             * exit immediately. If particle is on surface, we are assuming it's still inside and
+	     * if it goes outside(during next streaming step) we can backtrack it in the end of next streaming step.
              */
             const Scalar r2 = dot(pos,pos);
             const Scalar v2 = dot(vel,vel);
-            if (r2 <= m_R2 || v2 == Scalar(0))
+	    const Scalar v2_minus_V2 = v2 - m_V2;
```

Comment:
Both `v2` and `v2_minus_V2` can move after the conditional, since they don't need to be evaluated if this condition is met.

Reply:
Got it , Will do that